### PR TITLE
adding securityGroupResourceGroup to generated azure.json

### LINKED ISSUE #356: Cluster with NSG in separate RG fails to expose service
/kind bug

Because of https://github.com/kubernetes-sigs/cloud-provider-azure/issues/281, we are currently unable to specify a `securityGroupResourceGroup` in `azure.json` and the cloud-provider expects the NSG to be in the same RG as the cluster RG. Once the issue above is fixed, we should add `securityGroupResourceGroup` to all `azure.json` examples.

/assign

### PR DESCRIPTION
**Pull request by nader-ziada**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
- set securityGroupResourceGroup field to custom vnet resource group (if set) in generated azure.json

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #356 

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
 set securityGroupResourceGroup to custom vnet resource group in generated azure.json 
```

### CODE REVIEW

File: api/v1alpha3/types.go [hunk 1]
```diff
@@ -407,6 +407,10 @@ type SubnetSpec struct {
 	// +optional
 	SecurityGroup SecurityGroup `json:"securityGroup,omitempty"`
 
+	// The name of the resource group that the security group is deployed in
+	// +optional
+	SecurityGroupResourceGroup string `json:"securityGroupResourceGroup,omitempty"`
```
Comment (CecileRobertMichon) [inline api/v1alpha3/types.go:412]: I don't believe we want/need to expose this level of customization... In general I'd expect the security group to be in the same RG as the vnet or the rest of the cluster (or both). #1063 was mostly about adding `SecurityGroupResourceGroup:   d.Vnet().ResourceGroup` to support BYO vnet cases where the network infra is in a separate RG as the cluster (eg https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/356#issuecomment-735685949)

Comment (CecileRobertMichon) [reply api/v1alpha3/types.go:412]: I haven't heard of use cases from users that want their security group to be created for them in a different RG.

Comment (nader-ziada) [reply api/v1alpha3/types.go:412]: So just set the `security group resource group` to the `vnet resource group` if it’s different from the `cluster resource group`?

Comment (CecileRobertMichon) [reply api/v1alpha3/types.go:412]: Right. You can set it to always be vnet().ResourceGroup and that value will `== cluster resource group` when no vnet RG is specified

Comment (nader-ziada) [reply api/v1alpha3/types.go:412]: Oh ok, that easy to do, I misunderstood the requirement. Will make the changes.

Comment (nader-ziada) [reply api/v1alpha3/types.go]: @CecileRobertMichon made the changes, PTAL

**Review by CecileRobertMichon — APPROVED** [review]
/lgtm
/approve

**Pull request merged**